src/launch/nss-cache.h

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+/*
+ * NSS Cache
+ */
+
+#include <c-macro.h>
+#include <c-rbtree.h>
+#include <stdlib.h>
+#include <sys/types.h>
+
+typedef struct NSSCache NSSCache;
+
+enum {
+        _NSS_CACHE_E_SUCCESS,
+
+        NSS_CACHE_E_INVALID_NAME,
```

Comment:
"INVALID_NAME" sounds like the name was malformed, rather than just an unknown user.